frontend/src/core/user/actions.js

```diff
@@ -52,13 +64,21 @@ export function saveSetting(setting: string, value: boolean, username: string):
 
 
 /**
- * Sign out the current user.
+ * Mark all notifications of the current user as read.
+ *
+ * Once the API call completes, add class needed for CSS transition from
+ * `unread` to `default` state. The class gets removed after user data is
+ * updated, which happens after 1000ms - when transition is complete.
  */
-export function signOut(url: string): Function {
+export function markAllNotificationsAsRead(element: HTMLElement): Function {
     return async dispatch => {
-        await api.user.signOut(url);
+        await api.user.markAllNotificationsAsRead();
 
-        dispatch(get());
+        element.classList.add('fadeout');
+
+        setTimeout(function() {
+            dispatch(get());
+        }, 1000);
```

Comment:
Thanks, I thought there must be a better way to do this. :)

I followed your advice, but they way I understood and implemented it doesn't seem fruitful.

The initial state is set properly (unread notifications have lighter background).
It also updates when I close and re-open the menu (background changes to default color).
But transition does not happen after I open the menu for the first time.

Please help? :)

Reply:
Maybe I finally understand what you meant by "if you don't care to wait for the return value of the server". I pushed a potential solution, which simply performs the animation the first time the menu opens. 

I've also used CSS animation instead of transition, to make a slight delay.